module-system/sov-modules-api/src/lib.rs

```diff
@@ -305,6 +310,17 @@ pub trait Module {
     ) -> Result<CallResponse, Error> {
         unreachable!()
     }
+
+    /// Attempts to charge the provided amount of gas from the working set.
+    ///
+    /// The scalar gas value will be computed from the price defined on the working set.
+    fn charge_gas(
+        &self,
```

Comment:
We take `self` to allow the user to implement his own gas config, if he wants to

Reply:
Why do wee need this function in the `Context`? The caller can directly call `working_set.charge_gas(gas)`.